Source/monster.cpp

```diff
@@ -3987,15 +3988,12 @@ void StartMonsterDeath(Monster &monster, int pnum, bool sendmsg)
 	MonsterDeath(monster, pnum, md, sendmsg);
 }
 
-void M_StartKill(int monsterId, int pnum)
+void M_StartKill(Monster &monster, int pnum)
 {
-	assert(static_cast<size_t>(monsterId) < MaxMonsters);
-	Monster &monster = Monsters[monsterId];
-
 	if (pnum == MyPlayerId) {
-		delta_kill_monster(monsterId, monster.position.tile, *MyPlayer);
-		if (monsterId != pnum) {
-			NetSendCmdLocParam1(false, CMD_MONSTDEATH, monster.position.tile, monsterId);
+		delta_kill_monster(monster.getId(), monster.position.tile, *MyPlayer);

```

Comment:
Should cleanup the leafs first so we don't end up with a lot of ugly calls to `.getId()`

Reply:
For some reason, I thought that `delta_kill_monster` was going to remain the same. I will update it.

Reply:
it is monsterid, the golems have the same id as there player, which is how you look them up.

Reply:
Opened #4988

Reply:
Alright, it's rebased!